src/components/grouped-character/grouped-character-test.stories.tsx

```diff
@@ -77,3 +81,32 @@ NewValidation.args = {
   separator: "-",
   ...getCommonTextboxArgs(),
 };
+
+export const GroupedCharacterComponent = ({
+  onChange,
+  groups,
+  separator,
+  ...props
+}: GroupedCharacterProps) => {
+  const [state, setState] = React.useState("");
+
+  const setValue = ({ target }: CustomEvent) => {
+    setState(target.value.rawValue);
+    if (onChange) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      onChange(target);
```

Comment:
I would suggest doing this so we don't have to disable the type check for `onChange` :)

```suggestion
  const setValue = (ev: CustomEvent) => {
    setState(ev.target.value.rawValue);
    if (onChange) {
      onChange(ev);
```